src/context.jl

```diff
@@ -34,11 +34,6 @@ Behavior can be overridden by setting the `CLIMACOMMS_CONTEXT` environment varia
 to either `MPI` or `SINGLETON`.
 """
 function context(device = device(); target_context = context_type())
-    if target_context == :MPICommsContext && mpi_ext_is_not_loaded()
-        error(
-            "Loading MPI.jl is required to use MPICommsContext. You might want to call ClimaComms.@import_required_backends",
-        )
-    end
```

Comment:
Yes, but I don't feel that it is worth making this function depend on code loading logic (`get_extension`). I would rather we keep it simple, and document how this package should be used.

Reply:
`get_extension` is a really simple function that is probably compiled away and this code block is very clear. This error message is much more user and novice-friendly than "MPICommsContext has no methods".

I think that this is one of those cases where we can very precisely understand what the user's error was and be informative, instead of assuming that user read the documentation.

Reply:
It's not a simple function, and it cannot be compiled away, but I'd rather not waste more time on this, I'll leave it as it was.